applications/nrf5340_audio/src/audio/audio_system.c

```diff
@@ -576,6 +568,39 @@ int audio_system_decoder_num_ch_get(void)
 	return sw_codec_cfg.decoder.num_ch;
 }
 
+int audio_system_encoder_num_ch_set(uint32_t locations)
+{
+	if (IS_ENABLED(CONFIG_MONO_TO_ALL_RECEIVERS)) {
+		LOG_DBG("Cannot set encoder channels when MONO_TO_ALL_RECEIVERS is enabled");
+		return 0;
+	}
+
+	int num_ch = POPCOUNT(locations);
+
+	if (locations == BT_AUDIO_LOCATION_MONO_AUDIO) {
+		num_ch = 1;
+	}
+
+	if (num_ch > CONFIG_AUDIO_ENCODE_CHANNELS_MAX) {
+		LOG_ERR("Invalid number of encoder channels: %d (max %d)", num_ch,
+			CONFIG_AUDIO_ENCODE_CHANNELS_MAX);
+		return -EINVAL;
+	}
+
+	LOG_DBG("Setting encoder channels to %d based on locations 0x%08x", num_ch, locations);
+
+	sw_codec_cfg.encoder.num_ch = num_ch;
+	sw_codec_cfg.encoder.audio_loc = locations;
+
+	if (num_ch == 1) {
+		sw_codec_cfg.encoder.channel_mode = SW_CODEC_MONO;
+	} else {
+		sw_codec_cfg.encoder.channel_mode = SW_CODEC_MULTICHANNEL;
```

Comment:
This seems unnecessary with the latest changes. 
Can we have a loc structure instead of `channel_mode`?